File: Java/lab15_Command/task_2_1/src/Document.java
public class Document {

    private final String name;

    public Document(String name) {
        this.name = name;
        System.out.println("Creating new document " + name);
    }

    public void save() {
        System.out.println("Saving document " + name + "...");
    }

    public void showInfo() {
        System.out.println("Document " + this.name);
    }

    public void print() {
        System.out.println("Printing document " + name + "...");
    }
}

File: Cpp/lab15_Command/task_2_1/src/document.cpp
#include "document.hpp"

#include <iostream>


Document::Document(std::string const& name)
: name(name)
{
  std::cout << "Creating new document " << this->name << std::endl;
}

Document::~Document() {}

void Document::save()
{
  std::cout << "Saving document " << this->name << "..." << std::endl;
}

void Document::showInfo() const
{
  std::cout << "Document " << this->name << std::endl;
}

void Document::print() const
{
  std::cout << "Printing document " << this->name << "..." << std::endl;
}
